include/display.h

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2017 dXplore
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Public API for display drivers and applications
+ */
+
+#ifndef __DISPLAY_H__
+#define __DISPLAY_H__
+
+/**
+ * @brief Display Interface
+ * @defgroup display_interface display Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <device.h>
+#include <stddef.h>
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+enum display_pixel_format {
+	PIXEL_FORMAT_RGB_888 = 0,
+
+	PIXEL_FORMAT_END,
+};
+
+/**
+ * @typedef display_on_api
+ * @brief Callback API to turn display on
+ * See display_on() for argument description
+ */
+typedef int (*display_on_api)(struct device *dev);
+
+/**
+ * @typedef display_off_api
+ * @brief Callback API to turn display off
+ * See display_off() for argument description
+ */
+typedef int (*display_off_api)(struct device *dev);
+
+/**
+ * @typedef display_write_bitmap_api
+ * @brief Callback API for writing bitmap
+ * See display_write_bitmap() for argument description
+ */
+typedef int (*display_write_bitmap_api)(
+	const struct device *dev, const u16_t x, const u16_t y, const u16_t w,
+	const u16_t h, const enum display_pixel_format pixel_format,
+	const void *data);
+
+/**
+ * @typedef display_is_pixel_format_supported_api
+ * @brief Callback API to check if pixel format is supported by display
+ * See display_is_pixel_format_supported() for argument description
+ */
+typedef bool (*display_is_pixel_format_supported_api)(
```

Comment:
Moved from #9149 
> @pizi-nordic
> IMHO a bit mask might be better to list supported pixel formats. Then a "pixel format set" function should be available, as some devices are able to change pixel format. On the other hand, we might want a display_write() accepting multiple data formats, however I think that might be tricky for planar framebuffers.

Reply:
>On the other hand, we might want a display_write() accepting multiple data formats, however I think that might be tricky for planar framebuffers.

I see no reason to complicate the display driver.

Reply:
> .. Then a "pixel format set" function should be available, as some devices are able to change pixel format.  ...

I start wondering if this should not be all configured at compile time instead of adding code for dynamically switching at run time.